### PR TITLE
Release v1.0.0

### PR DESCRIPTION
**Pull request by gongled**
Initial release.

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -1 +1,136 @@
...
+	}
+
+	tpl.SetTemplate(options.GetS(OPT_TPL_PATH), name)
+	tpl.SetArgs(template.ParseArgsString(options.GetS(OPT_DEFINE)))
+
+	if err := tpl.RenderTemplate(wr); err != nil {
+		fmtc.Printf("Error: %s\n", err.Error())
+	}
+
+	defer wr.Close()
```
Comment (andyone) [inline cli/cli.go]: `defer` is useless there.

Comment (gongled) [reply cli/cli.go]: Fixed (fdc4d98709ba535a9205300af023b70b24961ab9).

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+########################################################################################
+
+# This Makefile generated by GoMakeGen 0.5.0 using next command:
```
Comment (andyone) [inline Makefile]: Try to regenerate this Makefile with latest version of GoMakeGen (0.6.0).

Comment (andyone) [reply Makefile]: And I highly recommend to use `--metalinter` option for generating metalinter validation target.

File: README.md [hunk 3]
```diff
@@ -25,15 +126,32 @@ Options
...
-### Documentation
+```
+Usage: kikky-cli {options}
 
-Read the documentation to get further information about mentioned software:
+Options
 
-- [Anaconda Documentation](https://fedoraproject.org/wiki/Anaconda)
-- [Kickstart Documentation](https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html)
+  --path, -p path ......... Path to templates directory
```
Comment (andyone) [inline README.md]: Looks like you use default help output. I recommend to use `--no-color` option for generating help content without dots.

File: cli/cli.go [hunk 4]
```diff
@@ -1 +1,136 @@
...
+func Init() {
+	opts, errs := options.Parse(optMap)
+
+	if len(errs) != 0 {
+		fmtc.Println("Errors while options parsing:")
+
+		for _, err := range errs {
+			fmtc.Printf("  %v", err)
+		}
+	}
```
Comment (andyone) [inline cli/cli.go]: You should exit from utility if package can't parse options.

File: cli/cli.go [hunk 5]
```diff
@@ -1 +1,136 @@
...
+
+	if options.GetB(OPT_HELP) {
+		showUsage()
+		os.Exit(0)
+	}
+
+	switch len(opts) {
+	case 0:
+		showUsage()
+		return
```
Comment (andyone) [inline cli/cli.go]: Useless `return`.

File: cli/cli.go [hunk 6]
```diff
@@ -1 +1,136 @@
...
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+// Process arguments
+func process(name string) {
+	tpl := template.KTemplate{}
+
+	wr, err := os.OpenFile(options.GetS(OPT_OUTPUT), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+
+	if err != nil {
+		wr = os.Stdout
```
Comment (andyone) [inline cli/cli.go]: Strange decision. You should print error about user _defined_ output file, not just use stdout.

File: cli/cli.go [hunk 7]
```diff
@@ -1 +1,136 @@
...
+	}
+
+	tpl.SetTemplate(options.GetS(OPT_TPL_PATH), name)
+	tpl.SetArgs(template.ParseArgsString(options.GetS(OPT_DEFINE)))
+
+	if err := tpl.RenderTemplate(wr); err != nil {
+		fmtc.Printf("Error: %s\n", err.Error())
```
Comment (andyone) [inline cli/cli.go]: You should print errors to stderr, not to stdout. And exit with exit code 1.

File: template/template.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func (t *KTemplate) formatPath(dir string) string {
+	return path.Join(dir, fmt.Sprintf("*.%s", K_EXT_NAME))
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func ParseArgsString(expr string) KArgs {
```
Comment (andyone) [inline template/template.go]: There is no docs at all in this file.

File: template/template.go [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+	var args = make(KArgs)
+
+	expr = strings.Replace(expr, "&", " ", -1)
+
+	for _, chunk := range strings.Split(expr, " ") {
+		pair := strings.Split(chunk, "=")
```
Comment (andyone) [inline template/template.go]: You can use [strutil.ReadField](https://godoc.org/pkg.re/essentialkaos/ek.v9/strutil#ReadField) for this. This function doesn't allocate additional memory, unlike `strings.Split`.

**Pull request merged**